### PR TITLE
Add export to file feature

### PR DESCRIPTION
**Pull request by rainqubit**
Hi Tantan, just wanna say i love your idea. I've been wanting to dip my hand into rust for a while now and after watching your vid i thought this is a good chance for me to do so. I'm still new to rust so i hope i implemented the feature correctly.

Preview:
aaa

Steps:
1. In `Command` mode press colon `:` to enter `CommandInput` mode,
2. In `CommandInput` mode press `W` then enter the path,
3. Press `enter` to save the image.

I'm using [image](https://docs.rs/image/0.23.8/image/) crate to encode the image.
ggez does provide encoding feature but only png format is currently supported so i choose image crate instead since it's support more formats.

I've only tested it on linux (Pop_OS 20.04) so i don't know if it'll work on windows or mac.

And also sorry, i butchered your code.

### CODE REVIEW

File: src/action_processer_utils.rs [hunk 1]
```diff
@@ -313,6 +315,21 @@ fn configure_edit_scale_uniform(action_processer_builder: &mut ActionProcesserBu
         );
 }
 
+pub fn configure_command_input(action_processer_builder: &mut ActionProcesserBuilder){
```
Comment (a-out-10-2) [inline src/action_processer_utils.rs:318]: Does this need to be public?

**Comment by a-out-10-2** [comment]
- Bug: Entering an invalid output filename; e.g. no filename suffix ("outfile"), filename only suffix (".png"), yields a 0-byte output file.
- Bug: InputVisualizer "Save Image:" prefix doesn't clear when the filename prompt is exited without entry.
- Comment: Errors seem to be printed to stdout.
- Huh: Exporting a file with a .jpeg or .jpg suffix appears to actually work. Verified with file headers: .png (89 50 4E 47) and jpg/jpeg (FF D8 FF E0).
- NBD/nitpick: Some of your commits overwrite each other.

File: src/action_processer_utils.rs (hunk 1 above)
Comment (rainqubit) [reply src/action_processer_utils.rs:318]: No, it doesn't. will change it to private

**Comment by rainqubit** [comment]
- For the filename bug, i'll add validation for it.
- Prefix bug, will fix it.
- Printing the error to stdout was intentional to help with debug/test, should i remove it?
- I've tested it with .png .jpg/.jpeg .gif and .bmp, here's the [list](https://docs.rs/image/0.23.8/image/#overview) of formats according to image crate.
- I'll try to make cleaner commits.

**Comment by a-out-10-2** [comment]
@rainqubit: how's it coming? I'm a test engineer by trade, but I don't need to be strict in this setting. @TanTanDev is the ultimate yea-er or nay-er.

**Comment by rainqubit** [comment]
Ah sorry, i haven't been able to get around and make the changes due to irl circumstances, i'll push the changes as soon as i can and it's ok if you're strict with me since i like to learn the best practices :)

**Comment by TanTanDev** [comment]
Sorry for taking such a long time to check this out, I've been working on many different projects, but got inspiration to finally check this pull request out.

As you saw I changed the input to go into CommandInput to 'C' from 'Colon' because it doesn't work on windows.
Thanks a bunch guys!

**Pull request merged**